Code/Framework/AzCore/AzCore/Serialization/Json/JsonImporter.cpp

```diff
@@ -144,21 +160,606 @@ namespace AZ
+                {
+                    return resolveResult;
+                }
+
+                if ((settings.m_resolveFlags & ImportTracking::Imports) == ImportTracking::Imports)
+                {
+                    rapidjson::Pointer path(element.Get().data(), element.Get().size());
+                    settings.m_importer->AddImportDirective(path, importName.String(), importAbsPath.String());
+                }
+                if ((settings.m_resolveFlags & ImportTracking::Dependencies) == ImportTracking::Dependencies)
+                {
+                    settings.m_importer->AddImportedFile(importAbsPath.String());
+                }
+
+                if (settings.m_resolveNestedImports)
+                {
+                    // Resolve the nested import by using the targetDataAtCurrentDepth as both the source
+                    // object to iterate over and the target object to copy to.
+                    // As the targetDataAtCurrentDepth has merged the current imported file,
+                    // the call will search for any $import directives in that imported file
+                    // and recursively resolve them
+                    ResultCode result = ResolveNestedImportsInOrder(targetDataAtCurrentDepth, allocator,
+                        targetDataAtCurrentDepth, importPathStack,
+                        settings, element, importAbsPath);
+                    if (result.GetOutcome() == Outcomes::Catastrophic)
+                    {
+                        return result;
+                    }
+                }
+            }
+            else
+            {
+                auto fieldNameStringRef = rapidjson::Value::StringRefType(fieldName.data(),
+                    static_cast<rapidjson::SizeType>(fieldName.size()));
+                resolveImportsResult.Combine(PatchField(targetDataAtCurrentDepth, allocator, source,
+                    fieldNameStringRef, *fieldValue, importPathStack, settings, element));
+            }
+        }
+
+        if (resolveImportsResult.GetProcessing() != Processing::Completed)
+        {
+            return resolveImportsResult;
+        }
+
+        // Move the local target data json value to the target result variable
+        target = AZStd::move(targetDataAtCurrentDepth);
+
+        return ResultCode(Tasks::Import, Outcomes::Success);
+    }
+
+    JsonSerializationResult::ResultCode JsonImportResolver::StoreNestedImports(const rapidjson::Value& jsonDoc,
+        ImportPathStack& importPathStack,
+        JsonImportSettings& settings, const AZ::IO::FixedMaxPath& importPath, StackedString& element)
+    {
+        using namespace JsonSerializationResult;
+
+        for (auto& path : importPathStack)
+        {
+            if (importPath == path)
+            {
+                return settings.m_reporting(
+                    AZStd::string::format("'%s' was already imported in this chain. This indicates a cyclic dependency.", importPath.c_str()),
```

Comment:
nit it looks like there are a few functions with ImportPathStack (489, 412) and some of the reporting uses c_str and others use `AZ_PATH_ARG` - not sure if it really matters, very minor thing.

Reply:
Anything that is using "%s" is using the `.c_str()` method to retrieve a null-terminated string.
Anything format that is using "%.*s" is using either an AZ::IO::PathView or AZStd::string_view which may not be null-terminated and therefore uses the AZ_STRING_ARG/AZ_PATH_ARG method.